plugins/inputs/suricata/suricata.go

```diff
@@ -101,7 +105,10 @@ func (s *Suricata) readInput(ctx context.Context, acc telegraf.Accumulator, conn
 			if rerr != nil {
 				return rerr
 			} else if len(line) > 0 {
```

Comment:
I think our linter will complain here. You should just have
```suggestion
			if rerr != nil {
				return rerr
			} 
			if len(line) > 0 {
```
as you exit on error anyway making the `else` superfluous.